bin/vault_sync_kv_credential.py

```diff
@@ -183,7 +203,14 @@ def stream_events(self, inputs, ew):
             self._logger.debug("{0}: found existing credential".format(input_name))
 
             found_credential = credential_session.storage_passwords[credential_title]
-            if found_credential.content.clear_password != fetched_vault_password:
+
+            found_clear_password = None
```

Comment:
Any particular reason for the `None` default versus maintaining uniform types with `""`?

Reply:
I was about to say there's no reason, and I like that suggestion.

But thinking through it, if I default to `""`, and the secret fetched from Vault is actually `""`, the TA would make no change to the stored credential, which would be the incorrect action.

I could be more specific in the later logic that [checks for equality](https://github.com/splunk/TA-VaultSync/pull/2/files/5d077a8de26f5c0682002f73a09d1cb727b00da6#diff-32e9e19de6edb662cb644a13a81f3f6e9774fa45f4c0aafb2ef7d930306aba68R213), and check for `== None or != fetched_vault_password`.  That seems mostly unnecessary, though.

Are there maintainability or other concerns around using `None` for what should eventually be set to a string? Because `None` is its own type, I think any use of it is likely to be non-uniform typing, so I'm not sure I see a reason to avoid it here.